x-pack/plugins/cases/public/components/configure_cases/index.tsx

```diff
@@ -114,8 +113,8 @@ export const ConfigureCases: React.FC = React.memo(() => {
   const onConnectorUpdated = useCallback<
     NonNullable<EditConnectorFlyoutProps['onConnectorUpdated']>
   >(
-    async (updatedConnector) => {
-      setEditedConnectorItem(updatedConnector as ActionConnectorTableItem);
+    async (updatedConnector: ActionConnector) => {
```

Comment:
nit: The callback parameter type is already properly inferred from the generic type:

<img width="820" alt="image" src="https://github.com/elastic/kibana/assets/18363145/c303b5a8-36d2-4032-aba4-73533d90ef97">